src/test/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/uiTest/SettingsTest.kt

```diff
@@ -0,0 +1,109 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.uiTest
+
+import com.intellij.remoterobot.RemoteRobot
+import nl.tudelft.ewi.se.ciselab.testgenie.uiTest.pages.IdeaFrame
+import nl.tudelft.ewi.se.ciselab.testgenie.uiTest.pages.SettingsFrame
+import nl.tudelft.ewi.se.ciselab.testgenie.uiTest.pages.WelcomeFrame
+import nl.tudelft.ewi.se.ciselab.testgenie.uiTest.utils.RemoteRobotExtension
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.MethodOrderer.OrderAnnotation
+import org.junit.jupiter.api.Order
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestMethodOrder
+import org.junit.jupiter.api.extension.ExtendWith
+import java.time.Duration
+
+@TestMethodOrder(OrderAnnotation::class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ExtendWith(RemoteRobotExtension::class)
+class SettingsTest {
+
+    @BeforeAll
+    fun setUpAll(remoteRobot: RemoteRobot): Unit = with(remoteRobot) {
+        find(WelcomeFrame::class.java, timeout = Duration.ofSeconds(15)).apply {
+            open("untitled")
+        }
+
+        Thread.sleep(10000)
```

Comment:
Generally, sleeping in tests is not good. Isn't there a way to wait until some element is visible instead?

Reply:
According to the following article, which explains how plugin testing can be done, the author explains there is no explicit way to set timeouts or make threads sleep: https://medium.com/@97malimi/testing-intellij-plugin-415c4ce014c6

Reply:
Sure, that makes sense.